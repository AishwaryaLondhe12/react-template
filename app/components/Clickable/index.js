/**
 *
 * Clickable
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { FormattedMessage as T } from 'react-intl'

const StyledClickable = styled.div`
  color: #1890ff;
  &:hover {
    cursor: pointer;
  }
`
function Clickable({ onClick, textId }) {
  return (
    <StyledClickable onClick={onClick}>
      {textId && <T id={textId} />}
    </StyledClickable>
  )
}

Clickable.propTypes = {
  onClick: PropTypes.func.isRequired,
  textId: PropTypes.string.isRequired
}

export default Clickable
