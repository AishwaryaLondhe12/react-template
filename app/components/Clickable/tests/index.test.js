/**
 *
 * Tests for Clickable
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
// import 'jest-dom/extend-expect' // add some helpful assertions

import Clickable from '../index'

Enzyme.configure({ adapter: new Adapter() })

describe('<Clickable />', () => {
  it('Should render and match the snapshot', () => {
    const component = shallow(
      <Clickable onClick={() => {}} textId="Clickable" />
    )
    expect(component).toMatchSnapshot()
  })
})
