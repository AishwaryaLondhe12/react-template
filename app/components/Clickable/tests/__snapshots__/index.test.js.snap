// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clickable /> Should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="Clickable__StyledClickable-sc-19lp5om-0 avquq"
      data-testid="clickable"
    />
  </div>
</body>
`;
